cluster-up/cluster/k8s-provider-common.sh

```diff
@@ -81,6 +76,11 @@ function up() {
 
     chmod u+x ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubectl
 
+    # Set server and disable tls check
+    export KUBECONFIG=${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubeconfig
+    ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubectl config set-cluster kubernetes --server="https://$(_main_ip):$(_port k8s)"
+    ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubectl config set-cluster kubernetes --insecure-skip-tls-verify=true
```

Comment:
We get `$kubectl`  variable from `cluster/ephemeral-provider-common.sh`, it is used in this file in other places too. Just changing that would be enough or you do need to move it down?

I've also noticed the `kubectl` usage on the `if podman` branch above, probably similar issue would happen there too

Reply:
Good catch.

There's definitely a chicken-and-egg problem here: in the case of podman, `${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubectl` (pointed to by the `$kubectl` variable) is downloaded for a specific k8s version in order to match the running cluster, but that in turn requires calling `kubectl`...

We could look into whether https://github.com/containers/conmon/issues/315, which is mentioned as the reason for this workaround existing, is still a problem in practice or whether we can just unify the code and avoid the additional call to `kubectl`.

Alternatively, it might be possible to obtain the value of `.status.nodeInfo.kubeletVersion` without needing `kubectl` by using `curl` against some HTTP endpoint? I'm not sure how feasible that would be.

Moving the calls down would still be necessary though, because if we left things as they are right now we would be calling `${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubectl` before having had the chance to create the file in question...

Reply:
So, on the podman topic, my guess is this has been fixed based on the comment... https://github.com/google/oss-fuzz/pull/4774#issuecomment-1308132677 

Looking at https://github.com/containers/conmon/issues/315#issuecomment-1027157864 seems that this was reproducible with conmon 2.1.0 but on the other comment it was fixed in 2.1.5

I _think_ we are doing some good tests with `podman` so if this breaks something, we will know quite fast too.